tools/eslint-rules/rules/folderStructureConfig.ts

```diff
@@ -0,0 +1,878 @@
+                    namePattern: 'camelCase',
+                    extension: ['tsx', 'ts'],
+                  },
+                },
+              ],
+            },
+            {
+              name: 'mock-data',
+              children: [
+                {
+                  name: 'generated',
+                  children: [
+                    {
+                      name: {
+                        namePattern: 'kebab-case',
+                        prefix: 'mock-',
+                        extension: 'ts',
+                      },
+                    },
+                  ],
+                },
+                {
+                  name: {
+                    namePattern: 'kebab-case',
+                    extension: 'ts',
+                  },
+                },
+                {
+                  name: {
+                    namePattern: 'camelCase',
+                    extension: 'ts',
+                  },
+                },
+                {
+                  name: {
+                    namePattern: 'camelCase',
+                    prefix: 'mock-',
+                    extension: 'ts',
+                  },
+                },
+              ],
+            },
+            {
+              name: 'profiling',
+              children: [
+                {
+                  name: 'components',
+                  children: [
+                    {
+                      name: {
+                        namePattern: 'StrictPascalCase',
+                        extension: 'tsx',
+                      },
+                    },
+                  ],
+                },
+                { ruleId: 'StatesFolder' },
+                { ruleId: 'ConstantsFolder' },
+                { ruleId: 'TypesFolder' },
+                { ruleId: 'UtilsFolder' },
+              ],
+            },
+            {
+              name: {
+                namePattern: 'StrictPascalCase',
+                extension: 'tsx',
+              },
+            },
+            {
+              name: {
+                namePattern: 'camelCase',
+                extension: ['tsx', 'ts'],
+              },
+            },
+          ],
+        },
+        // src/types
+        {
+          ruleId: 'TypesFolder',
+        },
+        // src/utils
+        { ruleId: 'UtilsFolder' },
+        // TODO: is it fine the way we are handling this?
+        {
+          name: {
+            namePattern: 'kebab-case',
+            extension: 'd.ts',
+          },
+        },
+        {
+          name: {
+            namePattern: 'index',
+            extension: 'css',
+          },
+        },
+        {
+          name: {
+            namePattern: 'index',
+            extension: 'tsx',
+          },
+        },
+        {
+          name: {
+            namePattern: 'App',
+            extension: 'tsx',
+          },
+        },
+        {
+          name: {
+            namePattern: 'SettingsRoutes',
+            extension: 'tsx',
+          },
```

Comment:
10. we could accept any file, but since there are few files, we are listing here